### ISSUE TITLE
The server goes into an endless loop when build ivf_sq8 index

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#3880]
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
Build ivf_sq8 index normally costs 35-60s for sift_10m(million), but this time it never ends.
The debug log always show this
```
[2020-09-26 11:24:57,123][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:24:57,125][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:24:58,132][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:24:58,134][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:24:59,134][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:24:59,136][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:25:00,134][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:25:00,136][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:25:01,135][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:25:01,137][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:25:02,138][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:25:02,140][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
[2020-09-26 11:25:03,140][DEBUG][ENGINE][BackgroundBuildIndexTask][build_index] Create BuildIndexJob for 91 segments of sift_10m_100000_128_l2
```
the info log:
```
[2020-09-26 11:09:52,080][INFO][SERVER][Cmd][grpcpp_sync_ser] Request [5] Cmd begin.
[2020-09-26 11:09:52,095][INFO][SERVER][Cmd][grpcpp_sync_ser] Request [5] Cmd end.
[2020-09-26 11:09:52,098][INFO][SERVER][CreateIndex][grpcpp_sync_ser] Request [6] CreateIndex begin.
[2020-09-26 11:12:47,818][INFO][SERVER][ShowCollections][grpcpp_sync_ser] Request [7] ShowCollectionss
 begin.
[2020-09-26 11:12:47,818][INFO][SERVER][ShowCollections][grpcpp_sync_ser] Request [7] ShowCollectionss
 end.
```

**Steps/Code to reproduce behavior**
build index yaml
```
build_performance:
  collections:
    -
      server:
        db_config.primary_path: /test/milvus/db_data_011/sift_10m_100000_128_l2_sq8_4096
        cache_config.cpu_cache_capacity: 32GB
        engine_config.use_blas_threshold: 1100
        engine_config.gpu_search_threshold: 1
        gpu_resource_config.enable: true
        gpu_resource_config.cache_capacity: 6GB
        gpu_resource_config.search_resources:
          - gpu0
          - gpu1
        gpu_resource_config.build_index_resources:
          - gpu1
      collection_name: sift_10m_100000_128_l2
      index_type: ivf_sq8
      index_param:
        nlist: 4096
```
 The athena has gpu0 and gpu1
1. drop original index and build index with gpu2
2. restart docker 
3. drop index and build index with gpu1

**Expected behavior**

**Method of installation**
- [ ] Docker/cpu
- [x] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)

- Milvus version (master or released version)
-0.11.0

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`
 The athena has gpu0 and gpu1
```yaml
gpu:
  enable: true
  cache_size: 5GB
  gpu_search_threshold: 1000
  search_devices:
    - gpu0
  build_index_devices:
    - gpu1

```

**Screenshots**

**Additional context**

**Comment by yhmo** [comment]
Could not reproduce in local

**Comment by yhmo** [comment]
pr3895 make a change to return build index error status to client

**Comment by yhmo** [comment]
#3895 has been merged into 0.11.0. 
In the latest build, specify non-exist gpu is not allowed, we can close this issue.
